server/rpc.go

```diff
@@ -283,8 +283,12 @@ func orchestratorInfo(orch Orchestrator, addr ethcommon.Address, serviceURI stri
 
 	os := drivers.NodeStorage.NewSession(authToken.SessionId)
 
-	if os != nil && os.IsExternal() {
-		tr.Storage = []*net.OSInfo{os.GetInfo()}
+	if os != nil {
+		if os.IsExternal() {
+			tr.Storage = []*net.OSInfo{os.GetInfo()}
+		} else {
+			os.EndSession()
```

Comment:
Why are we ending the session here?

Reply:
Because in this branch session object is not used, but it will remain in the session's map
https://github.com/livepeer/go-livepeer/blob/72eea36e9cca167c5a565b2384fc7cd2e3c1e533/drivers/local.go#L49
so it will be a memory leak.